SPHINXsys/src/shared/geometries/level_set.cpp

```diff
@@ -17,17 +17,25 @@ namespace SPH
 		}
 	}
 	//=================================================================================================//
-	Real BaseLevelSet::computeHeaviside(Real phi, Real half_width)
+	Real BaseLevelSet::Heaviside(Real phi, Real half_width)
 	{
 		Real heaviside = 0.0;
 		Real normalized_phi = phi / half_width;
 		if (phi < half_width && phi > -half_width)
-			heaviside = (0.5 + 0.5 * normalized_phi) + 0.5 * sin(Pi * normalized_phi) / Pi;
+			heaviside = 0.5 + 0.5 * normalized_phi;
 		if (normalized_phi > 1.0)
 			heaviside = 1.0;
 		return heaviside;
 	}
 	//=================================================================================================//
+	Real BaseLevelSet::computeVolumeFraction(Real cell_center_phi_, Real data_spacing, const Vecd position)
+	{
+		Vecd normal_direction = probeNormalDirection(position);
+		normal_direction /= normal_direction.norm() + TinyReal;
+		Real factor = 0.5 / normal_direction.cwiseAbs().maxCoeff();
+		return Heaviside(cell_center_phi_, factor * data_spacing);
```

Comment:
Calculation seems weird. If we take a normal along the cell diagonal [sqrt(2)/2,sqrt(2)/2] this does not work really well.

Reply:
I basically want to take the maximum absolute value of the vector. Do you have a proper way?

Reply:
Yes. We have developed the simple method which would yield the same result as we have introduced the contribution of normal direction. The idea is first determining the direction of plane them computing the cut.

Reply:
The simplified method yields quite different results from analytical. Why not simply computing the exact value when the plane actually cut the cell ?

Reply:
I have updated a more accurate version. In case you have even more accurate one, you can replace it.